edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -452,19 +454,19 @@ static async Task CloseDbStoreProviderAsync(IContainer container)
             await dbStoreProvider.CloseAsync();
         }
 
-        static Option<ulong> GetStorageMaxTotalWalSizeIfExists(IConfiguration configuration)
+        static Option<T> GetStorageConfigIfExists<T>(string fieldName, IConfiguration configuration)
         {
-            ulong storageMaxTotalWalSize = 0;
+            T storageParamValue = default(T);
             try
             {
-                storageMaxTotalWalSize = configuration.GetValue<ulong>(Constants.StorageMaxTotalWalSize);
+                storageParamValue = configuration.GetValue<T>(fieldName);
             }
             catch
             {
                 // ignored
```

Comment:
do we print out what storage configed value in log file?  and why you need to make it generic?